core/src/main/java/org/bitcoinj/core/LockTime.java

```diff
@@ -29,7 +31,37 @@
  * Instances of this class are immutable and should be treated as Java
  * <a href="https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/doc-files/ValueBased.html#Value-basedClasses">value-based</a>.
  */
-public class LockTime {
+public abstract /* sealed */ class LockTime {
+
+    public static final class HeightLock extends LockTime {
+
+        public HeightLock(long value) {
+            super(value);
+        }
+
+        public int blockHeight() {
+            return Math.toIntExact(value);
+        }
+    }
+
+    public static final class TimeLock extends LockTime {
+        public TimeLock(long value) {
+            super(value);
+        }
+
+        public Instant timestamp() {
+            return Instant.ofEpochSecond(value);
+        }
+    }
+
+    // TODO: Should we have a separate NoLock case, or should it be HeightLock with value ZERO?
```

Comment:
Since you're basing this PR on a commit with "unset == blockHeight 0", I'd pick the second choice.

Reply:
But you might be open to three subclasses in a subsequent PR?

Reply:
I'm reducing to two subtypes for this PR.